[Security Solution][Exceptions] - Fixes up some bugs in the all exception items view

yctercero opened the pull request:
## Summary

Addresses [#140709](https://github.com/elastic/kibana/issues/140709), [#141056](https://github.com/elastic/kibana/issues/141056), [#141421](https://github.com/elastic/kibana/issues/141421), [#141042](https://github.com/elastic/kibana/issues/141042)

-  Removed the option of selecting a single row - not sure there's much use there. 
<img width="535" alt="Screen Shot 2022-09-23 at 8 39 44 AM" src="https://user-images.githubusercontent.com/10927944/192016196-4190039f-b9f5-425e-a534-ad2e8dd80879.png">

- Updated text under exception tabs to indicate to user where each exception type is evaluated
<img width="767" alt="Screen Shot 2022-09-28 at 8 30 09 PM" src="https://user-images.githubusercontent.com/10927944/192933907-78b0ef8a-eda3-4803-a4fb-f220e5f67880.png">
<img width="690" alt="Screen Shot 2022-09-28 at 8 30 43 PM" src="https://user-images.githubusercontent.com/10927944/192933912-8182d91a-f91a-46c7-8814-5ae59069d0ff.png">

- Updated the exception lists management page search placeholder
<img width="373" alt="Screen Shot 2022-09-28 at 8 45 04 PM" src="https://user-images.githubusercontent.com/10927944/192933992-7da07452-13b9-4b7c-a9fd-09d7f41ead2d.png">

- Fixed some jank occurring on changing rows per page in the exception tabs

https://user-images.githubusercontent.com/10927944/192017552-9ad1acc9-fa91-4cae-90cd-702b6ba3cecc.mov


### Checklist

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


<!--ONMERGE {"backportTargets":["8.5"]} ONMERGE-->

yctercero committed:
fixing up some bugs in the all exception items view

elasticmachine commented:
Pinging @elastic/security-solution (Team: SecuritySolution)

yctercero commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into exceptions_postff_bugs

dhurley14 reviewed (approved):
Looks great thanks for the fixes!

xcrzx commented on x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/exceptions/translations.ts:
```diff
@@ -157,7 +157,7 @@ export const referenceErrorMessage = (referenceCount: number) =>
 export const EXCEPTION_LIST_SEARCH_PLACEHOLDER = i18n.translate(
   'xpack.securitySolution.exceptions.searchPlaceholder',
   {
-    defaultMessage: 'e.g. Example List Name',
```
As this message text changes, should we also remove the old text from the translation files?

yctercero replied:
I updated the key, so yea I think I'll need to remove the old one from the translation files.

xcrzx commented:
It looks like there's a bug somewhere in the exception pagination. I just opened a random rule and saw "Showing 1-0 of 0":

![Screenshot 2022-09-26 at 13 13 13](https://user-images.githubusercontent.com/1938181/192263267-57d89d47-af44-4e52-bdf4-e9342c46c3f0.png)

xcrzx commented on x-pack/plugins/security_solution/public/detection_engine/rule_exceptions/components/all_exception_items_table/index.tsx:
```diff
@@ -154,7 +160,16 @@ const ExceptionsViewerComponent = ({
+    useFindExceptionListReferences(exceptionListsToQuery);
+
+  useEffect(() => {
+    if (isLoadingReferences) {
+      setViewerState('loading');
+    } else {
+      setViewerState(null);
+    }
```
If reference loading has failed, should we set the viewer state to "error"?

nastasha-solomon commented:
@yctercero thanks for tagging the docs team for the UI copy review! Please find my comments and questions below.

Recommend capitalizing "id" in the placeholder text.
<img width="600" alt="exception-list-placeholder" src="https://user-images.githubusercontent.com/79124755/192351572-9105c513-34c6-4ca6-a2dc-e357ff7c84e2.png">

Couple notes/questions about this image:
- If the sentence at the top _always_ displays (i.e., it shows regardless of whether endpoint/rule exceptions exist) you can probably remove "Endpoint exceptions are applied to the endpoint and the rule detection" from the center text. 
-  Can we change the sentence at the top to align it with the language we use in our docs? For example, "Endpoint exceptions are added to both the detection rule and the Elastic Endpoint agent on your hosts." I feel like this explanation is pretty clear (shoutout to @joepeeples for the lang)
- Where is the second sentence linking to? In its current state, I'm not quite sure what it's recommending.
<img width="1486" alt="exception-tab-1" src="https://user-images.githubusercontent.com/79124755/192355069-57e16a93-9fc7-4ee7-9879-27f29454e01f.png">

A few suggestions for the text in this image:
- Swap "e.g." with "for example". We try to avoid using abbreviations like i.e. and e.g. to accomodate ESL folks.
- Is the query in the placeholder text KQL? If it is, what are your thoughts on the following revision:
Filter exceptions using Kibana Query Language (KQL), for example, name:"my list"
<img width="800" alt="exception-tab-2" src="https://user-images.githubusercontent.com/79124755/192358894-7761d13b-f885-4396-abd0-fed2302a6c66.png">

yctercero committed:
Merge branch 'main' of github.com:elastic/kibana into exceptions_postff_bugs

yctercero commented:
> @yctercero thanks for tagging the docs team for the UI copy review! Please find my comments and questions below.
> 
> Recommend capitalizing "id" in the placeholder text. <img alt="exception-list-placeholder" width="600" src="https://user-images.githubusercontent.com/79124755/192351572-9105c513-34c6-4ca6-a2dc-e357ff7c84e2.png">
> 

I updated it to specify that it was meant to be `list_id`, not the list `ID`. Does that make sense?

> Couple notes/questions about this image:
> 
> * If the sentence at the top _always_ displays (i.e., it shows regardless of whether endpoint/rule exceptions exist) you can probably remove "Endpoint exceptions are applied to the endpoint and the rule detection" from the center text.

Done!

> * Can we change the sentence at the top to align it with the language we use in our docs? For example, "Endpoint exceptions are added to both the detection rule and the Elastic Endpoint agent on your hosts." I feel like this explanation is pretty clear (shoutout to @joepeeples for the lang)
> * Where is the second sentence linking to? In its current state, I'm not quite sure what it's recommending.
> 
> <img alt="exception-tab-1" width="1486" src="https://user-images.githubusercontent.com/79124755/192355069-57e16a93-9fc7-4ee7-9879-27f29454e01f.png">
> 

Yea, I like that. I updated the text and removed that link - it was confusing. I was trying to give guidance as to how to link the endpoint list to the rule, but it's super not clear. 

> A few suggestions for the text in this image:
> 
> * Swap "e.g." with "for example". We try to avoid using abbreviations like i.e. and e.g. to accomodate ESL folks.
> * Is the query in the placeholder text KQL? If it is, what are your thoughts on the following revision:
>   Filter exceptions using Kibana Query Language (KQL), for example, name:"my list"
> 
> <img alt="exception-tab-2" width="800" src="https://user-images.githubusercontent.com/79124755/192358894-7761d13b-f885-4396-abd0-fed2302a6c66.png">

Updated the text! It's not KQL, it's [simple query syntax](https://www.elastic.co/guide/en/elasticsearch/reference/master/query-dsl-simple-query-string-query.html#simple-query-string-syntax) - I updated it to try to reflect that - let me know what you think.

yctercero committed:
apply feedback

yctercero committed:
Merge branch 'exceptions_postff_bugs' of github.com:yctercero/kibana into exceptions_postff_bugs

yctercero committed:
fix bug found in review around pagination showing when empty exception list linked to rule

yctercero commented:
> It looks like there's a bug somewhere in the exception pagination. I just opened a random rule and saw "Showing 1-0 of 0":
> 
> ![Screenshot 2022-09-26 at 13 13 13](https://user-images.githubusercontent.com/1938181/192263267-57d89d47-af44-4e52-bdf4-e9342c46c3f0.png)

Thanks for finding this! It's a result of an empty exception list being linked to a rule. Pushed up a fix.

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/precommit_hook.js --ref HEAD~1..HEAD --fix'

xcrzx reviewed (approved):
Thank you for the fixes, LGTM 👍

yctercero commented:
@nastasha-solomon I'm going to merge, but please still comment if changes are needed to copy and I will follow up with them.

nastasha-solomon commented on x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/exceptions/translations.ts:
```diff
@@ -155,9 +155,9 @@ export const referenceErrorMessage = (referenceCount: number) =>
+  'xpack.securitySolution.detectionEngine.rules.all.exceptions.searchPlaceholder',
   {
-    defaultMessage: 'e.g. Example List Name',
+    defaultMessage: 'Search by name or list_id',
```
Sorry for the delayed response. Feel free to revert this to `list id` or `list ID` so that it matches the way you're representing the `name` field in this helper text. That's all - everything else looks good. :) Thank you!

yctercero replied:
All good! I'll update and post follow up PR

yctercero replied:
Fixed here - https://github.com/elastic/kibana/pull/142258

nastasha-solomon reviewed (approved):
Just left one comment. Thanks!

kibanamachine commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.5|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/142237">](https://github.com/elastic/kibana/pull/142237)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

<!--- Backport version: 8.9.0 -->
